src/icrc1_auction_api_new.mo

```diff
@@ -474,17 +474,45 @@ actor class Icrc1AuctionAPI(quoteLedger_ : ?Principal, adminPrincipal_ : ?Princi
       case (null) null;
       case (?aid) getAssetId(aid);
     };
-    U.unwrapUninit(auction).getTransactionHistory(caller, assetId)
+    U.unwrapUninit(auction).getTransactionHistory(caller, assetId, #desc)
     |> U.sliceIter(_, limit, skip)
     |> Array.map<Auction.TransactionHistoryItem, TransactionHistoryItem>(_, func(x) = (x.0, x.1, x.2, Vec.get(assets, x.3).ledgerPrincipal, x.4, x.5));
   };
 
+  public shared query ({ caller }) func queryTransactionHistoryForward(token : ?Principal, limit : Nat, skip : Nat) : async ([TransactionHistoryItem], Nat, Bool) {
+    let assetId : ?Auction.AssetId = switch (token) {
+      case (null) null;
+      case (?aid) getAssetId(aid);
+    };
+    let a = U.unwrapUninit(auction);
+    let history = a.getTransactionHistory(caller, assetId, #asc)
+    |> U.sliceIter(_, limit, skip)
+    |> Array.map<Auction.TransactionHistoryItem, TransactionHistoryItem>(_, func(x) = (x.0, x.1, x.2, Vec.get(assets, x.3).ledgerPrincipal, x.4, x.5));
+
+    var sessionNumber : ?Nat = null;
+    var auctionInProgress : Bool = false;
+    for (aid in Vec.keys(assets)) {
+      switch (sessionNumber) {
+        case (null) sessionNumber := ?a.getAssetSessionNumber(aid);
+        case (?sn) {
+          let asn = a.getAssetSessionNumber(aid);
```

Comment:
Can you move this line to make it the first line after the `for` line. Between `for` and `switch`.

Reply:
done